src/libraries/Common/tests/Tests/Interop/osReleaseTests.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Text;
+using Xunit;
+
+namespace Common.Tests
+{
+    public class osReleaseTests
+    {
+        [Theory]
+        // Double quotes:
+        [InlineData("NAME=\"Fedora\"\nVERSION=\"37\"\nPRETTY_NAME=\"Fedora Linux 37\"", "Fedora Linux 37")]
+        [InlineData("NAME=\"Fedora\"\nVERSION=\"37\"", "Fedora 37")]
+        [InlineData("NAME=\"Fedora\"", "Fedora")]
+        // Single quotes:
+        [InlineData("NAME='Ubuntu'\nVERSION='22.04'\nPRETTY_NAME='Ubuntu Linux 22.04'", "Ubuntu Linux 22.04")]
+        [InlineData("NAME='Ubuntu'\nVERSION='22.04'", "Ubuntu 22.04")]
+        [InlineData("NAME='Ubuntu'", "Ubuntu")]
+        // No quotes:
+        [InlineData("NAME=Alpine\nVERSION=3.14\nPRETTY_NAME=Alpine_Linux_3.14", "Alpine_Linux_3.14")]
+        [InlineData("NAME=Alpine\nVERSION=3.14", "Alpine 3.14")]
+        [InlineData("NAME=Alpine", "Alpine")]
+        // No pretty name fields:
+        [InlineData("ID=fedora\nVERSION_ID=37", null)]
+        [InlineData("", null)]
+        public static void GetPrettyName_Success(
+            string content,
+            string expectedName)
+        {
+            string path = Path.GetTempFileName();
```

Comment:
We generally use FileCleanupTestBase or the utility class to create a test file. It puts it in a folder named for the test so you can see who leaked a file.

Reply:
I've changed to use `FileCleanupTestBase`, and applied the change also for `procfsTests`.